app/models/energy/ScheduleRuleset.py

```diff
@@ -0,0 +1,113 @@
+"""Schedule Ruleset Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+from app.models.energy.ScheduleBase import ScheduleContinuous, ScheduleDiscrete, ScheduleNumericType, ScheduleUnitType, DateTime, Date, Time
+
+
+class ScheduleTypeLimits (BaseModel):
+    """Specifies the data types and limits for values contained in schedules."""
+
+    type: Enum('ScheduleTypeLimits', {'type': 'ScheduleTypeLimits'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    lower_limit_value: float = Schema(
+        0,
+        description='Lower limit for the schedule type is entered.'
+    )
+
+    upper_limit_value: float = Schema(
+        1,
+        description='Upper limit for the schedule type is entered.'
+    )
+
+    numeric_type: ScheduleNumericType
+
+    unit_type: ScheduleUnitType
+
+
+class DayValue(BaseModel):
+    """Values for daily schedule"""
+
+    time: Time
+
+    value_until_time: float
+
+
+class ScheduleDay(BaseModel):
+    """Used to describe the daily schedule for a single simulation day."""
+    type: Enum('ScheduleDay', {'type': 'ScheduleDay'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    interpolate_to_timestep: bool
```

Comment:
I feel that there should be a default value here.  It's usually False in honeybee legacy so maybe we should just do it that way.